robottelo/hosts.py

```diff
@@ -569,55 +568,11 @@ def install_katello_host_tools(self):
         if result.status != 0:
             raise ContentHostError('Failed to install katello-host-tools')
 
-    def install_katello_ca(self, satellite):
-        """Downloads and installs katello-ca rpm on the content host.
-
-        :param str satellite: robottelo.hosts.Satellite instance
-
-        :return: None.
-        :raises robottelo.hosts.ContentHostError: If katello-ca wasn't
-            installed.
-        """
-        warnings.warn(
-            message='The install_katello_ca method is deprecated, use the register method instead.',
-            category=DeprecationWarning,
-            stacklevel=2,
-        )
-        self._satellite = satellite
-        self.execute(
-            f'curl --insecure --output katello-ca-consumer-latest.noarch.rpm \
-                    {satellite.url_katello_ca_rpm}'
-        )
-        # check if the host is fips-enabled
-        result = self.execute('sysctl crypto.fips_enabled')
-        if 'crypto.fips_enabled = 1' in result.stdout:
-            self.execute('rpm -Uvh --nodigest --nofiledigest katello-ca-consumer-latest.noarch.rpm')
-        else:
-            self.execute('rpm -Uvh katello-ca-consumer-latest.noarch.rpm')
-        # Not checking the status here, as rpm could be installed before
-        # and installation may fail
-        result = self.execute(f'rpm -q katello-ca-consumer-{satellite.hostname}')
-        # Checking the status here to verify katello-ca rpm is actually
-        # present in the system
-        if satellite.hostname not in result.stdout:
-            raise ContentHostError('Failed to download and install the katello-ca rpm')
-
-    def remove_katello_ca(self):
-        """Removes katello-ca rpm from the broker virtual machine.
-
-        :return: None.
-        :raises robottelo.hosts.ContentHostError: If katello-ca wasn't removed.
+    def reset_rhsm(self):
+        """Global Registration points the host's sub-man to talk to the Sattelite's Candlepin
+        but saves the original rhsm.conf. Reset the rhsm.conf so that it points back to the CDN.
         """
-        # unregister host from CDN to avoid subscription leakage
-        self.execute('subscription-manager unregister')
-        # Not checking the status here, as rpm can be not even installed
-        # and deleting may fail
-        self.execute('yum erase -y $(rpm -qa |grep katello-ca-consumer)')
-        # Checking the status here to verify katello-ca rpm is actually
-        # not present in the system
-        result = self.execute('rpm -qa |grep katello-ca-consumer')
-        if result.status == 0:
-            raise ContentHostError(f'katello-ca rpm(s) are still installed: {result.stdout}')
+        self.execute(r'\cp -f /etc/rhsm/rhsm.conf{.bak,}')
```

Comment:
I think when a host is never registered to any system other than CDN, where `/etc/rhsm/rhsm.conf.bak` won't exist, so I think it could fail here while copying, so can we add this to try and run sub-man clean in finally to avoid the file not present error?

Reply:
If the host was never registered to a SAT (and so the `rhsm.conf.bak` does not exist) the `cp` command would fail with status 1 and `self.execute` would just return the status and not throw any exception to be caught. Since we don't assert the status here (on purpose for this case) the code execution should not be interrupted.

```
In [5]: host.execute(r'\cp -f /etc/rhsm/rhsm.conf{.bak,}')
[DEBUG 240821 09:48:38] sat.redhat.com executing command: \cp -f /etc/rhsm/rhsm.conf{.bak,}
[DEBUG 240821 09:48:38] sat.redhat.com command result:
    stdout:
    
    stderr:
    cp: cannot stat '/etc/rhsm/rhsm.conf.bak': No such file or directory
    
    status: 1
Out[5]: 
stdout:

stderr:
cp: cannot stat '/etc/rhsm/rhsm.conf.bak': No such file or directory

status: 1
```

Reply:
yes, it would just fail silently and it won't throw any exception, However, I feel it would be better to handle it in a try-finally block to avoid any errors

Reply:
It is still unclear to me why you would like to handle it and what would be a part of the logical block that handles the non-zero return code. What is there to handle? I might be missing something.

Reply:
I think it's not needed. There is actually nothing we should do in case there is no `rhsm.conf.bak` (blank host will just keep pointing to the CDN),  so it doesn't make sense to me ho handle that return code. And since the `self.execute` does not throw any exception in that case I can see no reason to put it in the `try` block. Both of it just seem redundant to me and thus undesirable.